koku/masu/processor/ocp/ocp_report_db_cleaner.py

```diff
@@ -118,17 +120,36 @@ def purge_expired_report_data_by_date(self, expired_date, simulate=False):
                 all_cluster_ids.add(usage_period.cluster_id)
                 all_period_starts.add(str(usage_period.report_period_start))
 
-            LOG.info(
-                log_json(
-                    msg="removing all data related to cluster_ids",
-                    report_periods=all_report_periods,
-                    cluster_ids=all_cluster_ids,
-                    period_starts=all_period_starts,
-                    schema=self._schema,
+            if not simulate:
+                LOG.info(
+                    log_json(
+                        msg="removing all data related to cluster_ids",
+                        report_periods=all_report_periods,
+                        cluster_ids=all_cluster_ids,
+                        period_starts=all_period_starts,
+                        schema=self._schema,
+                    )
+                )
+                # Remove all data related to the report periods
+                del_count = OCPUsageLineItemDailySummary.objects.filter(
+                    report_period_id__in=all_report_periods
+                ).delete()
+                LOG.info(
+                    log_json(
+                        msg=f"Deleted records from table {OCPUsageLineItemDailySummary._meta.db_table} using id",
+                        count=del_count,
+                        schema=self._schema,
+                    )
+                )
```

Comment:
Are we sure this table has leftover data after data-expiration??? I thought we just needed to delete the old OCPUsageReportPeriod entries...

Reply:
I think so. Even though `OCPUsageLineItemDailySummary` model has [Cascade enabled](https://github.com/project-koku/koku/blob/afdbead2bd4ac4d66e5bc1e6bf635f1382cedee7/koku/reporting/provider/ocp/models.py#L120), the table definition has 

`REFERENCES <schema>.reporting_ocpusagereportperiod (id) MATCH SIMPLE`
`ON UPDATE NO ACTION`
 `ON DELETE NO ACTION`

(I think it should be ON DELETE CASCADE)

Which was raising `DETAIL:  Key (id)=(38) is still referenced from table "reporting_ocpusagelineitem_daily_summary".` That's why I suggested to delete from `OCPUsageLineItemDailySummary` too.

Reply:
I'm not following what you are saying. If we are having an issue with deleting from the `reporting_ocpusagereportperiod` because of the foreign key, then move the `OCPUsageReportPeriod.delete()` to below the data clean up in the PartitionTable. Like I mention above, we do not want to use the django delete method on a partition table when we are expiring data.

Reply:
Yes, what you've said makes complete sense. I moved `OCPUsageReportPeriod.delete()` down and removed `OCPUsageLineItemDailySummary` deletion. However, one point leaves me uncertain:

- Should we use `execute_delete_sql` for `OCPUsageReportPeriod` instead of Django's native `.delete()`,  or is this recommended only for partitioned tables? I tried that, but it didn’t work due to an FK constraint... and I noticed that in the table definition, there’s no ON CASCADE deletion (even though it is defined in the model). This is what confused me.